### PR TITLE
Documented withEntityNotFound api change

### PR DESCRIPTION
**Pull request by Kagaminara**
Related MR: https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/merge_requests/2354

[Preview](https://deploy-preview-734--heuristic-almeida-1a1f35.netlify.app/docs/remixed/migration/manual-migration)

### CODE REVIEW

File: docs/migration/manual-migration.mdx [hunk 1]
```diff
@@ -59,6 +58,37 @@ To cope with this change, you need to review all components using
...
+
+export default   withEntityNotFound({
+    isFound: () => false,
+    NotFoundComponent: NotFound,
+  })(MyComponent);
+
+```
+
+A even better way would be to get rid of `withEntityNotFound`, and instead
+handle this mechanism directly withing the related `loader` by throwing a `404`
```
Comment (mGGk-fr) [inline docs/migration/manual-migration.mdx:89]:
```suggestion
handle this mechanism directly within the related Remix route `loader` by throwing a `404`
```
Maybe provide an example of how to throw 404 ?

File: docs/migration/manual-migration.mdx [hunk 2]
```diff
@@ -59,6 +58,37 @@ To cope with this change, you need to review all components using
...
   ```
 
+## `withEntityNotFound` API change
+
+[`withEntityNotFound` API has changed](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/merge_requests/2354).
+In version 2, the parameter `NotFoundComponent` was optional (defaulted to
+`NotFound` component). However, this HOC is now deprecated and has been placed
+within `@front-commerce/compat` package, from which it is not possible to import
+files from the `theme`. Hence in v3, `withEntityNotFound`'s `NotFoundComponent`
+parameter is now mandatory.
```
Comment (real34) [inline docs/migration/manual-migration.mdx:68]: This is our internal implementation detail, I don't think it makes sense to expose all the reasons to readers.

**suggestion:** keep it simpler
```suggestion
`NotFound` component). This HOC is now deprecated and is now in the 
theme-agnostic `@front-commerce/compat` package.
If you still want to use `withEntityNotFound`, you must now provide a `NotFoundComponent`
component parameter.
```

File: docs/migration/manual-migration.mdx [hunk 3]
```diff
@@ -59,6 +58,37 @@ To cope with this change, you need to review all components using
...
+import NotFound "theme/modules/PageError/NotFound/NotFound"
+
+/*
+  ... your component
+*/
+
+export default   withEntityNotFound({
+    isFound: () => false,
+    NotFoundComponent: NotFound,
+  })(MyComponent);
```
Comment (real34) [inline docs/migration/manual-migration.mdx:84]:
```suggestion
export default withEntityNotFound({
    isFound: () => false,
    NotFoundComponent: NotFound,
})(MyComponent);
```

**Review by real34 — APPROVED** [review]

File: docs/migration/manual-migration.mdx (hunk 1 above)
Comment (Kagaminara) [reply docs/migration/manual-migration.mdx:89]: Added a link to Remix's guide on 404 ([this one](https://remix.run/docs/en/main/guides/not-found)).

**Pull request merged**